internal/data/spicedb.go

```diff
@@ -242,6 +244,54 @@ func (s *SpiceDbRepository) LookupResources(ctx context.Context, resouce_type *a
 	return resources, errs, nil
 }
 
+func (s *SpiceDbRepository) ImportBulkTuples(stream grpc.ClientStreamingServer[apiV1beta1.ImportBulkTuplesRequest, apiV1beta1.ImportBulkTuplesResponse]) error {
+	if err := s.initialize(); err != nil {
+		return err
+	}
+
+	var totalImported uint64
+	client, err := s.client.ImportBulkRelationships(context.Background())
+	if err != nil {
+		return fmt.Errorf("failed to create SpiceDB client: %w", err)
+	}
+
+	for {
+		req, streamErr := stream.Recv()
+		if streamErr != nil {
+			if req == nil && errors.Is(streamErr, io.EOF) {
+				if res, closeErr := client.CloseAndRecv(); closeErr != nil {
+					return fmt.Errorf("error receiving response from Spicedb for bulkimport request: %w", closeErr)
+				} else {
+					log.Infof("total number of relationships loaded: %d", res.NumLoaded)
+					totalImported = res.NumLoaded
+					return stream.SendAndClose(&apiV1beta1.ImportBulkTuplesResponse{NumImported: totalImported})
+				}
+			}
+			if !errors.Is(streamErr, io.EOF) {
+				if streamErr = stream.SendAndClose(&apiV1beta1.ImportBulkTuplesResponse{NumImported: totalImported}); err != nil {
+					return status.Errorf(codes.Internal, "failed to send response: %v", streamErr)
+				}
+			}
```

Comment:
Not sure about this one. It's setting `streamErr` but checking `err`? And looking at example gRPC client streaming server docs, it looks like they show if the error from the client `Recv()` is not EOF, they return that rather than SendAndClose. What do you think?

Reply:
yes, I agree.